### PR TITLE
LPS-182227 Create Tags Dropdown item on D&M

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
https://issues.liferay.com/browse/LPS-182227

It's under FF  LPS-84424

To test : 
- Go to Documents & Media
- Create files with some tags
- You can see the tags from the group

Problem:
on the figma it's asked this width:

image

but if I use the md size :

image

I think is something about the 
```
<clay:container-fluid>
	<liferay-ui:search-container
```
From the asset-tag-selector view
And I do not know if is ok as it is now or can be done something ( @jaimelr10 // @ambrinchaudhary )

- LPS-182227 add tags dropdown element
- LPS-182227 create modal with the tags contained

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js [hunk 1]
```diff
@@ -220,6 +220,18 @@ export default function propsTransformer({
 		});
 	};
 
+	const filterByTag = (tagsFilterURL) => {
+		openSelectionModal({
+			buttonAddLabel: Liferay.Language.get('select'),
+			height: '70vh',
+			multiple: true,
+			selectEventName: `${portletNamespace}selectedAssetTag`,
+			size: 'lg',
```
Comment (ambrinchaudhary) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js:229]: I would say that we should use the same modal size with than in filter by extension (`size: 'md'`)

**Review by ambrinchaudhary — CHANGES_REQUESTED** [review]
Hey @AliciaGarciaGarcia, not sure if your question is related about the modal size or the internal content size. This last one is provided by the content-dashboard view and we cannot change it. The modal size should be the same than for Filter by Extension. This is the size we should use even if does not correspond exactly with the designs (unless UX confirms that wants a specific width only for this case, which I would not suggest). 

So only a minor change requested. Also, I think you are missing the buildLang....

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java [hunk 2]
```diff
@@ -730,6 +783,8 @@ private long _getFileEntryTypeId() {
 	private List<DropdownItem> _getFilterNavigationDropdownItems() {
 		boolean extensionsIsEmpty = ArrayUtil.isEmpty(
 			_getExtensions(_httpServletRequest));
+		boolean tagsIsEmpty = SetUtil.isEmpty(
+			_getAssetTagIds(_httpServletRequest));
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:787]: Can you move this closer to its usage? If you can, I think it should be renamed to `active` (or inlined).

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java [hunk 3]
```diff
@@ -650,6 +660,49 @@ private void _addExtensionFilterLabelItems(
...
+		}
+
+		_assetTagIds = new HashSet(
+			Arrays.asList(
+				ArrayUtil.toStringArray(
+					ParamUtil.getStringValues(
+						httpServletRequest, "assetTagId"))));
+
+		return _assetTagIds;
+	}
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:675]: Shouldn't this be named `getSelectedTagNames` based on its usage?

Also, why are you calling `ArrayUtil.toStringArray`? `getStringValues` already returns a string array.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js:229]: I tried but it look like 
image

the search bar is missing.

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java (hunk 3 above)
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:675]: Even simpler, directly use `SetUtil.fromArray()`:

```java
		_assetTagIds = SetUtil.fromArray(
			ParamUtil.getStringValues(
				httpServletRequest, "assetTagId"));
```

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java (hunk 2 above)
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:787]: Yes, I would inline it.

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java (hunk 2 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:787]: this var will be used on posterior PRs as `extensionsIsEmpty` to know what the action is active. But I can inline it now, not problem.

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js (hunk 1 above)
Comment (ambrinchaudhary) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js:229]: Ok, after seeing this with you via slack, I agree you should keep the size='lg', as content-dashboard does the same, and we cannot change the internal view.

**Review by ambrinchaudhary — APPROVED** [review]

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js:229]: thanks!

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by marcogalluzzi — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java [hunk 4]
```diff
@@ -900,6 +950,19 @@ private long _getRepositoryId() {
...
 	}
 
+	private Set<String> _getSelectedAssetTagIds(
+		HttpServletRequest httpServletRequest) {
+
+		if (_assetTagIds != null) {
+			return _assetTagIds;
+		}
+
+		_assetTagIds = SetUtil.fromArray(
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:960]: A very small thing: this field name should be renamed too to be consistent with the getter.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java (hunk 4 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:960]: Mmmmmm
The problem here is that we doing 
`ParamUtil.getStringValues(httpServletRequest, "assetTagId")`

don't you think that request name has the priority on the naming?

Comment (adolfopa) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:960]: Ok, as tests have passed let's forward it and cross fingers.

**Review by adolfopa — APPROVED** [review]

**Pull request closed without merging**